extensions/proposals/WEBGL_video_texture/extension.xml

```diff
@@ -52,13 +52,11 @@ interface WebGLVideoFrameInfo {
 };
 
 [NoInterfaceObject]
-interface OES_EGL_image_external {
-    const GLenum TEXTURE_EXTERNAL_OES             = 0x8D65;
-    const GLenum SAMPLER_EXTERNAL_OES             = 0x8D66;
-    const GLenum TEXTURE_BINDING_EXTERNAL_OES     = 0x8D67;
-    const GLenum REQUIRED_TEXTURE_IMAGE_UNITS_OES = 0x8D68;
+interface WEBGL_video_texture {
+    const GLenum TEXTURE_VIDEO_IMAGE             = 0x851D;
+    const GLenum SAMPLER_VIDEO_IMAGE             = 0x8B61;
 
-    [RaisesException] WebGLVideoFrameInfo EGLImageTargetTexture2DOES(
+    [RaisesException] WebGLVideoFrameInfo VideoElementTargetVideoTexture(
```

Comment:
Yes, after this call returns, the texture is guaranteed to be filled with the video frame, and the WebGLVideoFrameInfo contains the metadata for that frame. It's a synchronous API.

The Promise-based API you suggested above would be asynchronous, requiring the developer to wait for the Promise to be resolved before they could work with the texture.